bidi-scanner/cli/tests/cli.test.mjs

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { resolve, join, dirname } from 'path';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { jest } from '@jest/globals';
+import { fileURLToPath } from 'url';
+import { processFiles } from '../cli.mjs';
+
+const filename = fileURLToPath(import.meta.url);
+const thisDirectory = dirname(filename);
+
+const mockLogger = {
+  fatal: jest.fn(),
+  error: jest.fn(),
+  warn: jest.fn(),
+  info: jest.fn(),
+  debug: jest.fn(),
+  trace: jest.fn(),
+  child: () => mockLogger,
+  mockClear: () => {
+    mockLogger.fatal.mockClear();
+    mockLogger.error.mockClear();
+    mockLogger.warn.mockClear();
+    mockLogger.info.mockClear();
+    mockLogger.debug.mockClear();
+    mockLogger.trace.mockClear();
+  },
+};
+
+describe('when testing for bidirectional (bidi) characters', () => {
+  describe('given the input directory does not exist', () => {
+    let exitCode = 0;
+
+    beforeAll(async () => {
+      mockLogger.mockClear();
+
+      exitCode = processFiles(mockLogger, ['-d', 'fake']);
```

Comment:
Mocking with ES modules is a bit harder, and I was running into difficulties even when I "did it correctly." So I opted finally for dependency injection, passing the logger into the functions that need access to it.